src/lib/utils/createSendEventForHits.ts

```diff
@@ -0,0 +1,157 @@
+import { AlgoliaSearchHelper } from 'algoliasearch-helper';
```

Comment:
Why are these files not in the Insights middleware folder?

Reply:
These files are quite big enough by themselves including test cases (the test cases are added in #4464).
They are there because of the middlewares, but also they're not directly used by middlewares but by connectors. So I put them in separate files.